.github/workflows/refapp-3x-login.yml

```diff
@@ -1,17 +1,43 @@
 name: RefApp 3.x Login
 on:
+  push:
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
+  repository_dispatch:
+    types: [qa]
+  workflow_dispatch:
 jobs:
   build:
     runs-on: ubuntu-latest
+    services:
+      firefox:
+        image: selenium/node-firefox:3.141.59-gold
+        env:
+          HUB_HOST: hub
+          HUB_PORT: 4444
     steps:
-      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Checkout qaframework
+        uses: actions/checkout@master
+        with:
+          repository: ${{github.repository}}
+      - name: run db and web containers
+        run: |
+          docker-compose -f docker/docker-compose-refqa.yml up -d
+      - name: wait for openmrs instance to start
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 1; done
       - name: Using Node.js
         uses: actions/setup-node@v1
         with:
           node-version: '12.x'
       - name: Installing dependencies
         run: npm install
-      - name: Run patient registration workflow tests
+      - name: Run RedApp 3.x frontend
+        run: npx openmrs@next start --backend localhost:8080 &
```

Comment:
So here's where things might get a little bit difficult. If the backend is already running on port 8080, then we can't have the frontend running on port 8080. Only one executable can claim one port at a time. So, either we need a new docker-compose file that runs the backend on a port other than 8080 (e.g., 8081) or we need to run the frontend on a port other than 8080 and ensure that Cypress connects to that port.

Reply:
PS The way we make things work in the 3.x distro is precisely this. We have three web Docker components:

* The 3.x UI running on one server
* The OpenMRS backend running on another server
* An entrance proxy running on a third server, which keeps everything tied to the same port.

You can see the details of how this works by examining the setup [here](https://github.com/openmrs/openmrs-distro-referenceapplication/tree/3.x/run/docker). The key file for making everything appear to be running on the same port is [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/3.x/run/docker/proxy/proxy.conf). Note how the frontend and backend both run on 8080, but on different "hostnames".

Reply:
Oh! Thanks, Ian. This is very helpful!

I tried running the proxy on a different port. The UI was there but it didn't load any data.
the following error occurs, but let me update the workflow and try it.
```
[HPM] Error occurred while proxying request localhost:8081/openmrs/ws/rest/v1/session to localhost:8080 [EHOSTUNREACH] (https://nodejs.org/api/errors.html#errors_common_system_errors)
[HPM] Error occurred while proxying request localhost:8081/openmrs/ws/rest/v1/session to localhost:8080 [EHOSTUNREACH] (https://nodejs.org/api/errors.html#errors_common_system_errors)
[HPM] Error occurred while proxying request localhost:8081/openmrs/ws/rest/v1/visit?includeInactive=false&v=custom:(uuid,patient:(uuid,identifiers:(identifier,uuid),person:(age,display,gender,uuid)),visitType:(uuid,name,display),location:(uuid,name,display),startDatetime,stopDatetime) to localhost:8080 [EHOSTUNREACH] (https://nodejs.org/api/errors.html#errors_common_system_errors)
```

Reply:
@ian @brandones,
I changed the frontend to run in port 8081 (`--port 8081`) also changed the port in cypress.
Still no luck.